src/models/user.rs

```diff
@@ -20,3 +25,55 @@ pub struct NewUser {
     pub name: Option<String>,
     pub avatar: Option<String>,
 }
+
+// Request guard for user authorization
+#[rocket::async_trait]
+impl<'r> FromRequest<'r> for User {
+    type Error = ();
+
+    async fn from_request(req: &'r Request<'_>) -> request::Outcome<Self, Self::Error> {
+        let cookies = req.cookies();
+
+        // Check Authorization header and cookies
+        let token = req
+            .headers()
+            .get_one("Authorization")
+            .and_then(|e| e.strip_prefix("Bearer "))
```

Comment:
I think I missed this initially but we should enforce that the header starts with `Bearer`, not just strip off the prefix (so if it *doesn't* start with that, we should ignore it)

Reply:
`strip_prefix` actually fails if the string doesn't have the prefix :yeah: